### PR TITLE
4.x: Fix timed delay hangs with dotCover and the DefaultScheduler

### PR DESCRIPTION
**Pull request by akarnokd**
For some reason, dotCover would hang in various timed test which use the `DefaultScheduler`. I don't know why, but removing that empty `try-catch` along with the atomic dispose management solves the issue. 

In addition, the `Virtual_ThreadSafety()` was running extremely slow in its original form (more than 5 minutes). The reduced time amount and a more close thread-synchronization should be faster now while preserving the behavior under test.

### CODE REVIEW

File: Rx.NET/Source/src/System.Reactive/Concurrency/ConcurrencyAbstractionLayerImpl.cs [hunk 1]
```diff
@@ -183,23 +174,16 @@ private static void Tick(object state)
...
-
             public void Dispose()
             {
-                var timer = _timer;
-                if (timer != TimerStubs.Never)
+                if (Disposable.TryDispose(ref _timer))
                 {
                     _action = Stubs<object>.Ignore;
-                    _timer = TimerStubs.Never;
                     _state = null;
```
Comment (danielcweber) [inline Rx.NET/Source/src/System.Reactive/Concurrency/ConcurrencyAbstractionLayerImpl.cs]: I see a race here where _state could be set null but the timer will execute, passing the wrong state (it doesn't help that _action is set to Stubs.Ignore before, that might be rearranged). Maybe timer._action should only be called when TryDispose was true, setting _action and _state by itself (or by some DisposeCore method)?

**Comment by danielcweber** [comment]
Do you know anything about the issue or bug here? Seems that the SpinWait would loop forever?

**Comment by akarnokd** [comment]
I don't know, I can't switch to debug while running with coverage enabled. Maybe there is a way to trigger a thread dump externally?

**Comment by danielcweber** [comment]
I could give it a try.

**Comment by danielcweber** [comment]
Do you know what process will eventually host our code? Everything I debug doesn't run anything form System.Reactive.

**Comment by danielcweber** [comment]
I can't reproduce the hanging tests under coverage. `Virtual_ThreadSafety()` running slow is understandable, it takes a lot of loops and `AdvanceBy` to actually trigger the virtual scheduler.

**Comment by akarnokd** [comment]
Were you running dotCover or microsoft's own coverage tool of VS Enterprise?

**Comment by danielcweber** [comment]
DotCover within VS

**Comment by akarnokd** [comment]
There could be a probabilistic signal loss from the timer.

You could try and add a loop around the body of `Delay_TimeSpan_DefaultScheduler` for example, usually 1.000 - 10.000 iterations. 

```cs
        [Fact]
        public void Delay_TimeSpan_DefaultScheduler()
        {
            for (int i = 0;  i < 10000; i++)    
                Assert.True(Observable.Return(1).Delay(TimeSpan.FromMilliseconds(1))
                   .ToEnumerable().SequenceEqual(new[] { 1 }));
        }
```

**Comment by danielcweber** [comment]
Can't really reproduce it but the fix is straightforward and the SpinWait-solution is wacky enough that this should be merged.

**Pull request merged**